presto-orc/src/main/java/com/facebook/presto/orc/OrcCorruptionException.java

```diff
@@ -20,18 +20,23 @@
 public class OrcCorruptionException
         extends IOException
 {
-    public OrcCorruptionException(String message)
+    public OrcCorruptionException(OrcDataSourceId orcDataSourceId, String message)
     {
         super(message);
     }
 
-    public OrcCorruptionException(String messageFormat, Object... args)
+    public OrcCorruptionException(OrcDataSourceId orcDataSourceId, String messageFormat, Object... args)
     {
-        super(format(messageFormat, args));
+        super(formatMessage(orcDataSourceId, messageFormat, args));
     }
 
-    public OrcCorruptionException(Throwable cause, String messageFormat, Object... args)
+    public OrcCorruptionException(Throwable cause, OrcDataSourceId orcDataSourceId, String messageFormat, Object... args)
     {
-        super(format(messageFormat, args), cause);
+        super(formatMessage(orcDataSourceId, messageFormat, args), cause);
+    }
+
+    private static String formatMessage(OrcDataSourceId orcDataSourceId, String messageFormat, Object[] args)
+    {
+        return "Malformed ORC file " + orcDataSourceId + ". " + format(messageFormat, args);
```

Comment:
Maybe format with brackets: `Malformed ORC file [hdfs://abc/xyz]. Invalid file size: 1234`

Could also put it at the end:
```
Malformed ORC file. Invalid file size: 1234 (hdfs://abc/xyz)
Malformed ORC file. Invalid file size: 1234 [hdfs://abc/xyz]
Malformed ORC file. Invalid file size: 1234 (file: hdfs://abc/xyz)
```

If you think the current way looks best, that's fine too.